### PR TITLE
Prepare for declaring OTLP/JSON Stable

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This adds guarantees that are necessary for OTLP/JSON wire representation stability.

This does not yet declare OTLP/JSON. It will be a separate PR once we are certain nothing else is needed.

### CODE REVIEW

**Review by reyang — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: README.md
Comment (bogdandrutu) [inline README.md:59]: Can you clarify why here by adding a link to the specification.

**Review by bogdandrutu — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

File: README.md
Comment (tigrannajaryan) [reply README.md:59]: Sorry, I don't understand. What do you want to link to and from where?

Comment (bogdandrutu) [reply README.md:59]:
```suggestion
- Enum choice names may change. This is allowed because enum names are not sent on the wire.
```

Comment (tigrannajaryan) [reply README.md:59]: Done

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers last call to review before I merge this.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
